frontend/src/app/services/user.service.ts

```diff
@@ -13,7 +13,17 @@ export class UserService {
   private currentUser?: string = undefined;
   private currentUserIsAdmin?: boolean = false;
 
-  constructor(private authService: AuthService, private router: Router, private userApiService: UsersApiService) { }
+  constructor(private authService: AuthService, private router: Router, private userApiService: UsersApiService) { 
+    const token = localStorage.getItem("authorizationToken");
+    if (token) {
+      this.authService.testAuthentication(token).subscribe(res => {
+        if (res) {
```

Comment:
res is always true in this context, you should check for "res.success"

Reply:
Roger.